test/dates.test.js

```diff
@@ -93,3 +93,36 @@ test("dates are prettified as expected", () => {
   expect(prettifyDate("MONTH", "2020-01-01")).toStrictEqual("2020-Jan");
   expect(prettifyDate("CENTURY", "-3000-01-01")).toStrictEqual("-3000"); // BCE
 });
+
+
+const ambiguousCalendarDates = {
+  "2024-XX-XX": 2024.5,
+  "2024-01-XX": 2024.044,
+  "2024-11-XX": 2024.88, 
+  "2024-12-XX": 2024.96,
+}
+
+test("calendarToNumeric doesn't allow ambiguous dates unless requested", () => {
+  for (const [calendarDate, _] of Object.entries(ambiguousCalendarDates)) {
+    expect(calendarToNumeric(calendarDate)).toBe(undefined)
+  }
+});
+
+test("calendarToNumeric allows ambiguous dates when requested", () => {
+  for (const [calendarDate, numericDate] of Object.entries(ambiguousCalendarDates)) {
+    expect(calendarToNumeric(calendarDate, true)).toBeCloseTo(numericDate);
+  }
+});
+
+const invalidCalendarDates = [
+  "string",
+  "-string",
+  "2024-XX-01",
+  2024.123,
+]
```

Comment:
```suggestion
  "2024-X-XX",
  "2024-XXX-XX",
]
```

Reply:
Thanks - this comment made me realise I wasn't testing dates which were ambiguous + erroneous correctly. We now have two separate tests:
* calendarToNumeric returns undefined for erroneous dates without ambiguity
* calendarToNumeric returns undefined for erroneous dates with ambiguity